### PR TITLE
Add support for VkPipelineFragmentShadingRateStateCreateInfoKHR

### PR DESCRIPTION
**Pull request by hakzsam**

### CODE REVIEW

File: fossilize.cpp [hunk 1]
```diff
@@ -995,6 +999,24 @@ static void hash_pnext_struct(const StateRecorder *,
 	h.u32(info.dstAccessMask);
 }
 
+static void hash_pnext_struct(const StateRecorder *,
+                              Hasher &h,
+                              const VkPipelineFragmentShadingRateStateCreateInfoKHR &info,
+			      const DynamicStateInfo *dynamic_state_info)
```
Comment (HansKristian-Work) [inline fossilize.cpp:1005]: Weird indent.

File: fossilize.cpp [hunk 2]
```diff
@@ -995,6 +999,24 @@ static void hash_pnext_struct(const StateRecorder *,
...
+	if (dynamic_state_info && !dynamic_state_info->fragment_shading_rate)
+	{
+		h.u32(info.fragmentSize.width);
+		h.u32(info.fragmentSize.height);
+		h.u32(info.combinerOps[0]);
+		h.u32(info.combinerOps[1]);
+	}
+	else
+	{
+		h.u32(0);
```
Comment (HansKristian-Work) [inline fossilize.cpp:1016]: Don't need to hash in 0 here. sType is already hashed in automatically.

File: fossilize.cpp [hunk 3]
```diff
@@ -7130,6 +7205,30 @@ static bool json_value(const VkMemoryBarrier2KHR &create_info, Allocator &alloc,
 	return true;
 }
 
+template <typename Allocator>
+static bool json_value(const VkPipelineFragmentShadingRateStateCreateInfoKHR &create_info, Allocator &alloc, Value *out_value,
+		       const DynamicStateInfo *dynamic_state_info)
```
Comment (HansKristian-Work) [inline fossilize.cpp:7210]: Indent. Alignment is with spaces.

File: fossilize.cpp (hunk 1 above)
Comment (hakzsam) [reply fossilize.cpp]: Fixed.

File: fossilize.cpp (hunk 3 above)
Comment (hakzsam) [reply fossilize.cpp]: Fixed.

File: fossilize.cpp (hunk 2 above)
Comment (hakzsam) [reply fossilize.cpp]: Fixed.

**Review by HansKristian-Work — APPROVED** [review]
LGTM

**Pull request merged**